feat(MarketsOffers): Implement the market offers table.

abouolia opened the pull request.

abouolia committed:
feat(MarketsOffers): Implement the market transactions table.

abouolia committed:
chore(Markets): Styling markets transactions table.

abouolia committed:
chore: Implementing the markets offers page.

abouolia committed:
chore(MarketOffers): Add submit hooks.

abouolia committed:
chore: Abstruct table style to variant.

abouolia committed:
chore(MarketTransactions): Add some assertions to the component.

abouolia committed:
chore(MarketsOffers): Add Spanish localization.

abouolia committed:
chore(WalletOffers): WIP

abouolia commented on packages/renderer/src/components/organisms/MarketsTransactions/MarketsTransactions.tsx:
```diff
@@ -0,0 +1,44 @@
+import { MarketTransactionsTable } from "@molecules/MarketTransactionsTable";
+import type { FC } from "react";
+
+export function MarketsTransactionsLoaded() {
```
Replace to `MarketsOffersLoaded`.

abouolia committed:
chore: Rename from transactions to offers.

abouolia committed:
chore: Add percentage component.

abouolia committed:
chore: fix notes.

abouolia committed:
chore: use color from theme colors schema.

abouolia committed:
Merge branch 'develop' into ahmed/feature/market-offers

abouolia committed:
chore: fix eslint errors.

abouolia committed:
fix: update snapshot tests.

schowdhuri commented on packages/renderer/src/components/atoms/Currency/Currency.tsx:
```diff
@@ -15,15 +15,19 @@
+
+interface CurrencyProps extends FormatNumberOptions {
   currencyCode?: string;
+  currentDisplay?: CurrencyFormatType;
```
`currentDisplay` isn't a good choice for a name. How about `format`?

abouolia replied:
This is outdated, i fixed that already.

schowdhuri commented on packages/renderer/src/components/atoms/Currency/Currency.tsx:
```diff
@@ -15,15 +15,19 @@
+
+interface CurrencyProps extends FormatNumberOptions {
   currencyCode?: string;
+  currentDisplay?: CurrencyFormatType;
```
`currentDisplay` isn't a good choice for a name. How about `format`?

abouolia replied:
I don't know how did that. 🤦 🤦

schowdhuri commented on packages/renderer/src/components/atoms/Percentage/Percentage.tsx:
```diff
@@ -0,0 +1,23 @@
+}
+
+export function Percentage({ value }: PercentageProps) {
+  return <>%{value * 100}</>;
```
Is this the correct format? Percentage values are supposed to look like `3.45%`, `25%`, etc

schowdhuri commented on packages/renderer/src/components/atoms/Percentage/Percentage.tsx:
```diff
@@ -0,0 +1,23 @@
+}
+
+export function Percentage({ value }: PercentageProps) {
+  return <>%{value * 100}</>;
```
1. Is this the correct format? Percentage values are supposed to look like `3.45%`, `25%`, etc.
2. this doesn't need to be a component. A util method like `fractionToPercent` is far simpler and more descriptive

abouolia replied:
Agree with you, Removed percentage as a component and relayed on a small utility. btw all percentages data should be stored as fractions.

schowdhuri commented on packages/renderer/src/components/molecules/MarketOffersTable/MarketOffersTable.tsx:
```diff
@@ -0,0 +1,144 @@
+const table = createTable().setRowType<MarketOffer>();
+
+interface MarketOffersTableProps {
+  data: MarketOffer[];
```
Global comment: `Array<Foo>` notation. This has been a recurring issue. Please make a note of this for all future work.

abouolia replied:
Remembered all the recurring mistakes except this one. :D

schowdhuri commented on packages/renderer/src/components/molecules/MarketOffersTable/MarketOffersTable.tsx:
```diff
@@ -0,0 +1,144 @@
+const table = createTable().setRowType<MarketOffer>();
+
+interface MarketOffersTableProps {
+  data: MarketOffer[];
```
Global comment: `Array<Foo>` notation. This has been a recurring issue. Please make a note for future work.

schowdhuri commented on packages/renderer/src/components/organisms/MarketsOffers/_utils.ts:
```diff
@@ -0,0 +1,36 @@
+import type { MarketOffer } from "@molecules/MarketOffersTable";
+
+export const transformToMarketsOffers = (
+  offers: MarketOfferData[]
```
`Array<Foo>` notation

schowdhuri replied:
`Array<Foo>` notation

schowdhuri commented on packages/renderer/src/components/organisms/MarketsOffers/_utils.ts:
```diff
@@ -0,0 +1,36 @@
+
+export const transformToMarketsOffers = (
+  offers: MarketOfferData[]
+): MarketOffer[] => {
```
`Array<Foo>` notation

schowdhuri replied:
`Array<Foo>` notation

schowdhuri commented on packages/renderer/src/components/organisms/MarketsOffers/_utils.ts:
```diff
@@ -0,0 +1,36 @@
+    amount: offer.amount,
+    amountCurrency: offer.baseCurrencyCode,
+    costCurrency: offer.baseCurrencyCode,
+    priceComparison: 0.1,
```
Are these constants: `priceComparison`, `cost`?

schowdhuri commented on packages/renderer/src/components/organisms/MarketsOffers/_utils.ts:
```diff
@@ -0,0 +1,36 @@
+    amount: offer.amount,
+    amountCurrency: offer.baseCurrencyCode,
+    costCurrency: offer.baseCurrencyCode,
+    priceComparison: 0.1,
```
Are these constants: `priceComparison`, `cost`?

abouolia replied:
Ok mapped the cost to `txFees` but we haven't price comparison or any percentage value.

schowdhuri commented on packages/renderer/src/components/templates/NavbarLayout/index.tsx:
```diff
@@ -14,15 +14,27 @@
+interface NavbarLayoutProps extends DefaultProps {
+  children: ReactNode;
+}
+export const NavbarLayout = ({
```
Use `FC` and remove `  children: ReactNode`

abouolia replied:
`NavbarLayoutProps ` extends to `DefaultProps`, so `NavbarLayout ` has more than `children` prop.

schowdhuri replied:
```tsx
const NavbarLayout: FC<DefaultProps> = (props) => {
 // ...
}
```

schowdhuri commented on packages/renderer/src/components/templates/NavbarLayout/index.tsx:
```diff
@@ -14,15 +14,27 @@
+interface NavbarLayoutProps extends DefaultProps {
+  children: ReactNode;
+}
+export const NavbarLayout = ({
```
Newline required

schowdhuri commented on packages/renderer/src/hooks/haveno/useMarketsOffers.ts:
```diff
@@ -0,0 +1,82 @@
+export function useMarketsOffers(query: MarketsOfferesQuery) {
+  const client = useHavenoClient();
+
+  return useQuery<MarketOfferData[], Error>(
```
`Array<Foo>` notation

schowdhuri replied:
`Array<Foo>` notation

schowdhuri commented on packages/renderer/src/hooks/haveno/useMarketsOffers.ts:
```diff
@@ -0,0 +1,82 @@
+  );
+}
+
+const transformData = (offers: OfferInfo[]) => {
```
`Array<Foo>` notation

schowdhuri replied:
`Array<Foo>` notation

schowdhuri reviewed (changes requested):
- remove Account Age and Number of trades from the table
- clicking on the `Market` link doesn't do anything
- let's upgrade `MarketOffersTable` to an organism
- `Array<Foo>` notation must be used everywhere

abouolia committed:
chore: Rewrite [] array to Array<>.

abouolia committed:
fix: format currency.

abouolia committed:
chore: Move MarketOffersTable under organisms.

abouolia committed:
fix: PR notes.

abouolia committed:
chore: FIx PR notes.

abouolia committed:
chore: add copyright licenses.

abouolia commented:
- Hided the Account Age and Number of trades columns from the table.
- fixed: clicking on the Market link doesn't do anything.
- fixed: let's upgrade MarketOffersTable to an organism.
- fixed: Array<Foo> notation must be used everywhere

abouolia committed:
chore.

schowdhuri reviewed (changes requested):
Thanks! This is nearly done. Just one tiny change and we'll be good to go!

abouolia committed:
chore(NavbarLayout): Fixed TS error.

schowdhuri reviewed (approved):
Looks perfect! 🚀
